symbology/read-symbols-from-mobile-style-file/src/main/java/com/esri/samples/read_symbols_from_mobile_style_file/SymbolLayerInfoListCell.java

```diff
@@ -33,29 +27,28 @@ protected void updateItem(SymbolStyleSearchResult item, boolean empty) {
       // if the item in the list view is an empty item, show nothing
       setGraphic(null);
     } else {
-      ImageView symbolImageView = new ImageView();
+      var symbolImageView = new ImageView();
       setGraphic(symbolImageView);
       // get the symbol from the list view entry, and create an image from it
-      ListenableFuture<Symbol> symbolFuture = item.getSymbolAsync();
-      symbolFuture.addDoneListener(() -> {
-        try {
-          // get the resulting symbol
-          Symbol symbol = symbolFuture.get();
-          // create a bitmap swatch from the symbol
-          ListenableFuture<Image> imageListenableFuture = symbol.createSwatchAsync(Color.TRANSPARENT, 1);
-          imageListenableFuture.addDoneListener(() -> {
-            try {
-              Image symbolImage = imageListenableFuture.get();
-              // update the image view with the symbol swatch
-              symbolImageView.setImage(symbolImage);
-            } catch (InterruptedException | ExecutionException e) {
-              new Alert(Alert.AlertType.ERROR, "Error creating preview image for symbol in mobile style file" + e.getMessage()).show();
-            }
-          });
-        } catch (InterruptedException | ExecutionException e) {
-          new Alert(Alert.AlertType.ERROR, "Error getting symbol" + e.getMessage()).show();
-        }
-      });
+      item.getSymbolAsync().toCompletableFuture().whenComplete(
+        (symbol, ex) -> {
+          if (ex == null) {
+            // create a bitmap swatch from the symbol
+            symbol.createSwatchAsync(Color.TRANSPARENT, 1).toCompletableFuture()
+              .whenComplete((symbolImage, e) -> {
+                if (e == null) {
+                  // update the image view with the symbol swatch
+                  symbolImageView.setImage(symbolImage);
```

Comment:
Is the flickering fixed or was that a different sample?

Reply:
Unfortunately not fixed - I've left the offending method (`ReadSymbolsFromMobileStyleFileController.updateSymbolPreview`) using a ListenableFuture for now, which avoids the issues but isn't a particularly satisfying conclusion.

Reply:
I suggest taking this sample out of this PR and I'll create a separate issue for it as it sounds like we are better off not patching it with completable futures and then leaving that bit with a ListenableFuture if it wasn't quite right in the first place with needing to use that `get`. Let's explore it separately and see if completable futures can work after a bit of a rewrite.